core/src/test/scala/neotypes/TransactIntegrationSpec.scala

```diff
@@ -1,33 +1,63 @@
 package neotypes
 
-import neotypes.implicits.mappers.results._
+import neotypes.implicits.mappers.all._
 import neotypes.implicits.syntax.string._
-import org.scalatest.Assertion
-
+import neotypes.internal.syntax.async._
+import org.neo4j.driver.v1.exceptions.ClientException
+import org.scalatest.compatible.Assertion
 import scala.concurrent.Future
 import scala.reflect.ClassTag
 
-/** Base class for integration specs that test the Session.transact method. */
+/** Base class for testing the Session[F].transact method. */
 abstract class TransactIntegrationSpec[F[_]] extends CleaningIntegrationSpec[F] {
+  import TransactIntegrationSpec.CustomException
+
   def fToFuture[T](f: F[T]): Future[T]
 
-  final def ensureCommitedTransaction[T](expectedResults: T)
-                                        (txF: Transaction[F] => F[T])
-                                        (implicit F: Async[F]): Future[Assertion] =
-    fToFuture(execute(s => s.transact(txF))).map {
-      results => assert(results == expectedResults)
-    }
+  implicit def F: Async[F]
 
-  final def ensureRollbackedTransaction[E <: Throwable : ClassTag]
-                                       (txF: Transaction[F] => F[Unit])
-                                       (implicit F: Async[F]): Future[Assertion] =
+  private final def ensureCommitedTransaction[T](expectedResults: T)
+                                                (txF: Transaction[F] => F[T]): Future[Assertion] =
+    fToFuture(
```

Comment:
I like the changes to this trait to unify the query logic in one place to reduce the duplication!  That's nice!  One small comment on the api, is that scala.Future is a supported effect type, but seems to have more influence than then other effect types, since in the fToFuture method we provide an implementation to convert the other effect types into Future.  I understand that this is primarily due to the influence exerted by AsyncFlatSpec, since it likes a Future[Assertion] but doesn't know what to do about F[Assertion].  One alternative would be to re-implement fToFuture to something like evaluateF and the new method evaluateF would return a T instead of a Future[T].

Reply:
The thing is that it would make the tests slower and sometimes will not work, because they will block.
Also, this isn't really nothing outside the standard, most projects using **IO** or other effects write their tests using `usafeRunToFuture` and the `AsyncFlatSpec`.